### ISSUE TITLE
Deprecate lambdify decorator

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1820]
### Relevant Discussion

https://github.com/pyccel/pyccel/discussions/1819#discussioncomment-8996708

### Describe the feature
The `decorators.py` file contains an unused decorator `@lambdify`. This decorator is designed to operate on lambda functions, but lambda functions shouldn't need such a decorator. Its existence also makes it harder to understand the use case for other decorators as its name is similar to the role of other decorators. It should therefore be removed.

cc: @saidctb

**Comment by saidctb** [comment]
We should raise a `NotImplemented` error when the user uses a lambda function until we decide if we want to support it

**Comment by EmilyBourne** [comment]
> We should raise a `NotImplemented` error when the user uses a lambda function until we decide if we want to support it

I agree but I am not sure what the relevance is to this issue. Please create a dedicated issue

**Comment by EmilyBourne** [comment]
> > > We should raise a `NotImplemented` error when the user uses a lambda function until we decide if we want to support it
> > 
> > 
> > I agree but I am not sure what the relevance is to this issue. Please create a dedicated issue
> 
> It is related, lambdify only works on lambda functions, if you remove lambdify you need to handle lambdas

At the moment `lambdify` doesn't work at all even if it was originally conceived to work on lambdas.

But actually a dedicated issue is not needed as this is already the behaviour in Pyccel at the moment. The following code:
```python
a = lambda x: x+1

b = a(2)
```
gives:
```
ERROR at annotation (semantic) stage
pyccel:
 |fatal [semantic]: tmp_lambdas.py [1,4]| Lambda functions are not currently supported ((FunctionDefArgument('x'),) -> x + 1)
```

**Comment by EmilyBourne** [comment]
Fixed by #1823
